torchao/quantization/quant_api.py

```diff
@@ -550,6 +552,7 @@ def apply_float8wo_quant(weight):
 def float8_dynamic_activation_float8_weight(
     activation_dtype: torch.dtype = torch.float8_e4m3fn,
     weight_dtype: torch.dtype = torch.float8_e4m3fn,
+    granularity: Literal[PerTensor, RowWise] = PerTensor,
```

Comment:
@jerryzh168 @vkuzo Thoughts on what the public API should be with this? I stole the granularity from the observers module. Note I could also do: PerAxis(0) as the only supported option

Reply:
would be good to have granularity for each tensor (for example, rowwise for input1 and tensorwise for input2).